kratos/geometries/geometry.h

```diff
@@ -634,6 +634,99 @@ class Geometry
         return mData.GetValue(rThisVariable);
     }
 
+    ///@}
+    ///@name Dynamic access to internals
+    ///@{
+
+    /* Set s a certain value to the geometry,
+     * according to a variable.
+     * Allows dynamic interfaces with each respective geometry.
+     */
+
+    /// Set with bool
+    virtual void Set(
+        const Variable<bool>& rVariable,
+        bool& Output) {}
+
+    /// Set with int
+    virtual void Set(
+        const Variable<int>& rVariable,
+        int& Output) {}
+
+    /// Set with double
+    virtual void Set(
+        const Variable<double>& rVariable,
+        double& Output) {}
+
+    /// Set with array_1d<double, 2>
+    virtual void Set(
+        const Variable<array_1d<double, 2>>& rVariable,
+        array_1d<double, 2>& Output) {}
+
+    /// Set with array_1d<double, 3>
+    virtual void Set(
+        const Variable<array_1d<double, 3>>& rVariable,
+        array_1d<double, 3>& Output) {}
+
+    /// Set with array_1d<double, 6>
+    virtual void Set(
+        const Variable<array_1d<double, 6>>& rVariable,
+        array_1d<double, 6>& Output) {}
```

Comment:
```suggestion
    /// Set with array_1d<double, 3>
    virtual void Set(
        const Variable<array_1d<double, 3>>& rVariable,
        array_1d<double, 3>& Output) {}
```
I would keep the extension separate

Reply:
To be honest, right now, I am explicitly after the array_1d<double, 2>. I'd actually like to keep it.